File: binance4j-websocket/src/main/java/com/binance4j/websocket/client/WebsocketClient.java
package com.binance4j.websocket.client;

import com.binance4j.websocket.configuration.WebsocketClientConfiguration;

/**
 * A websocket client
 */
public interface WebsocketClient {

	/**
	 * Opens the stream
	 */
	void open();

	/**
	 * Forces the closing of the stream
	 */
	void close();

	/**
	 * The client configuration
	 */
	WebsocketClientConfiguration getConfiguration();
}

File: connectors/src/main/kotlin/com/binance4j/websocket/client/WebsocketClient.kt
package com.binance4j.websocket.client

import com.binance4j.websocket.configuration.WebsocketClientConfiguration

/** A websocket client  */
interface WebsocketClient {
    /** Opens the stream.  */
    fun open()

    /** Forces the closing of the stream.  */
    fun close()

    /**
     * Closes of the stream
     *
     * @param force force the closing?
     */
    fun close(force: Boolean)

    /** @return The client configuration.
     */
    val configuration: WebsocketClientConfiguration?
}